lib/convert.js

```diff
@@ -160,37 +156,58 @@
         // Write an image of format <format> to STDOUT
         args.push(format + ":-");
 
-        // Setup a file stream to the output file
-        var fileStream = fs.createWriteStream(path);
-        fileStream.on("error", function (err) {
-            fileCompleteDeferred.reject("Could not create write stream for file " + path + ": " + err);
+        return args;
+    }
+
+    function _shouldUsePNGQuant(settings) {
+        return settings.usePngquant && settings.format === "png" && settings.pngquantQuality === 8;
+    }
+
+    function _pipeThroughPNGQuant(binaryPaths, inputStream, outputStream, outputCompleteDeferred) {
+        var pngquantProc = spawn(binaryPaths.pngquant, ["-"]);
+
+        pngquantProc.on("error", function (err) {
```

Comment:
Refactoring looks good. I think you can take it one step further and take this block and the block at 196 and turn that into a function too and be done with it all.